lib/decompress/zstd_decompress.c

```diff
@@ -2179,7 +2179,9 @@ size_t ZSTD_decompressStream(ZSTD_DStream* zds, ZSTD_outBuffer* output, ZSTD_inB
         case zdss_flush:
             {   size_t const toFlushSize = zds->outEnd - zds->outStart;
                 size_t const flushedSize = ZSTD_limitCopy(op, (size_t)(oend-op), zds->outBuff + zds->outStart, toFlushSize);
-                op += flushedSize;
+
+                op = op ? op + flushedSize : op;
```

Comment:
If `op == NULL` here,
then is was already `NULL` at previous line,
where it's used within `(oend-op)` operation,
which already does not seem proper.

Therefore, if `op==NULL`, my suggestion would be to skip the whole flush operation,
which is understandable : there is simply no room to flush anything.

In which case, the whole block could become conditional, starting with a `if (op != NULL)`.
When `op==NULL`, it jumps straight to the end: `someMoreWork = 0; break;`.

The point is less about speed optimization, than about clarity of situation : 
inside the `{ ... }` code block, there is now a guarantee that `op != NULL`.